pypika/tests/test_selects.py

```diff
@@ -1187,6 +1187,21 @@ def test_join_with_with(self):
               str(test_query),
         )
 
+    def test_join_with_with_insert_returning(self):
```

Comment:
I guess the name should be a bit more descriptive and should also describe the fact that we are testing the alias that was a problem before.

Reply:
If the part that was broken is the subquery having an alias, then the test should be named like that. Adjust to the actual problem solved accordingly.

Reply:
It is the `.returning('*')` statement in subquery that is actually broken, not an alias. If we omit it, everything works as expected. If we use it outside of subquery, everything works as expected. Though I'm not sure if it affects only `WITH` queries or inlined subqueries are affected as well (such as `SELECT FROM (INSERT ...)`).

I've simplified the test query a bit and renamed the test, would you mind taking another look at it?

Reply:
This is perfect, thanks! It's now a lot more clear what the bug is that gets fixed.